app/models/product.rb

```diff
@@ -215,12 +215,19 @@ def stage
     end
 
     def write_manifest
+      manifest = Manifest.load( router.path.join('manifest.yml') )
+
+      if manifest.valid?
+        manifest.name = title
+        manifest.description = description
+      else
+        manifest = Manifest.new({ "name" => title, "description" => description })
+      end
+
       File.open(router.path.join('manifest.yml'), 'w') do |f|
-        f.write({
-          'name' => title,
-          'description' => description
-        }.to_yaml)
+        f.write(manifest.to_yaml)
       end if (!title.blank? || !description.blank?) || !router.path.join('manifest.yml').exist?
+
       true
     end
```

Comment:
After suggested changes the code would be:

```ruby

# OodApp#manifest_path should be created

Manifest.load(app.manifest_path).merge(name: title, description: description).save(manifest_path) if (!title.blank? || !description.blank?) || !app.manifest_path.exist?
```

If statement will go away after fixing https://github.com/OSC/ood-dashboard/issues/87

* save should throw exception and we should handle it here